lib/compress/zstd_compress.c

```diff
@@ -2388,7 +2388,8 @@ ZSTD_buildSequencesStatistics(seqStore_t* seqStorePtr, size_t nbSeq,
                         const ZSTD_fseCTables_t* prevEntropy, ZSTD_fseCTables_t* nextEntropy,
                               BYTE* dst, const BYTE* const dstEnd,
                               ZSTD_strategy strategy, unsigned* countWorkspace,
-                              void* entropyWorkspace, size_t entropyWkspSize) {
+                              void* entropyWorkspace, size_t entropyWkspSize,
+                              const U32 selectEncodingTypes) {
```

Comment:
I'm annoyed that this `selectEncodingTypes` concept gets externalized in the function prototype.

It looks to me that it would make more sense to keep these topics separated.

What you could have is something like 
`ZSTD_symbolEncodingTypeStats_t ZSTD_buildDummyStatistics()`
and it would do the same thing as what's inside the `if (selectEncodingTypes) { } else { ... } ` statement.

It would then be selected directly within `ZSTD_buildBlockEntropyStats_sequences()`, instead of `ZSTD_buildSequencesStatistics()`, depending on `nbSeq==0` or not.